vars/sectionDeployToAKS.groovy

```diff
@@ -19,9 +20,10 @@ def testEnv(String testUrl, block) {
   }
 }
 
-def clearHelmReleaseForFailure(AppPipelineConfig config, DockerImage dockerImage, Map params, PipelineCallbacksRunner pcr) {
-  if (config.clearHelmReleaseOnFailure) {
-    helmUninstall(dockerImage, params, pcr)
+def clearHelmReleaseForFailure(boolean enableHelmLabel, AppPipelineConfig config, DockerImage dockerImage, Map params, PipelineCallbacksRunner pcr) {
+  def projectBranch = new ProjectBranch(env.BRANCH_NAME)
+  if((projectBranch.isPR() && !enableHelmLabel) || (projectBranch.isMaster() && config.clearHelmReleaseOnFailure)) {
```

Comment:
Aren't we removing `clearHelmReleaseOnFailure` so this would prevent this working without the change?
```suggestion
  if ((projectBranch.isPR() && !enableHelmLabel) || projectBranch.isMaster()) {
```

Reply:
Just speaking to Narinder, we thought we still need`clearHelmReleaseOnFailure` flag for master as there are no labels an teams may want a way to debug / troubleshoot staging failures.

Reply:
Yes we probably still need that, config will need to be changed though so that this is the default value. I wonder if it should have a deprecation flag to alert teams they have it enabled too or just leave it initially?

Reply:
we can probably leave it as is, as the behaviour of it is not changing for master ie. don't delete if the flag isn't set ?

Reply:
shouldn't we delete failures by default? the logs are captured by Jenkins.